src/main/java/com/kodilla/ecommercee/mapper/ProductMapper.java

```diff
@@ -0,0 +1,36 @@
+package com.kodilla.ecommercee.mapper;
+
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.domain.ProductDto;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ProductMapper {
+
+    public Product mapToProduct(final ProductDto productDto) {
+        return new Product(
+                productDto.getProductId(),
```

Comment:
Podczas createProduct podjąc dto nie znasz id więc nie możesz go tutaj użyć. Z drugiej strony używasz to też przy update i wtedy znasz id. Lepiej mapper zrobić bez id, a przy update użyć settera, aby zaktualizować obiekt.

Reply:
Dokonałem zmian, choć nie do końca jestem do nich przekonany.